gen/calc.py

```diff
@@ -575,6 +575,30 @@ def fz_err(msg):
             validate_int_in_range(port, 1, 65535)
 
 
+def calculate_scarce_resources(enable_scarce_resources):
+    return ','.join(enable_scarce_resources)
+
+
+def calculate_filter_gpu_resources(enable_scarce_resources):
```

Comment:
Since `enable_scarce_resources` is a data structure that comes straight from user config. it's a json-encoded string. You need to `json.loads()` it before inspecting it. As it's currently written, this function is checking string length.